streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -203,8 +208,10 @@ public void cleanRemovedTasks() {
             if (!locks.containsKey(id)) {
                 try {
                     if (lock(id, 0)) {
-                        log.info("Deleting obsolete state directory {} for task {}", dirName, id);
-                        Utils.delete(taskDir);
+                        if (taskDir.lastModified() < time.milliseconds() - cleanupDelayMs) {
```

Comment:
better: `if (time.milliseconds() > taskDir.lastModified() + cleanupDelayMs) { `
(easier to parse for humans, IMHO)